File: ZenChattyServer.Net/Models/GroupSettings.cs
using System.ComponentModel.DataAnnotations;

namespace ZenChattyServer.Net.Models;

public class GroupSettings()
{
    [Key]
    public Guid Id { get; set; } = Guid.NewGuid();
    
    [MaxLength(80)]
    public string DisplayName { get; set; } = "Group";
    
    [MaxLength(80)]
    public string AvatarFileLocator { get; set; } = "";
    public bool IsAllSilent { get; set; } = false;
    public bool IsInviteOnly { get; set; } = false; // invite link, QR code with expiration // todo: deprecated: no need to allow group search for now (time cost)
    public bool IsPrivateChatAllowed { get; set; } = false;
}

File: front.expo.rn/src/class/GroupSettings.ts
export default class GroupSettings
{
    displayName: string;
    isAllSilent: boolean;
    isInviteOnly: boolean; // invite link, QR code with expiration 
    isPrivateChatAllowed: boolean;
    
    constructor(
        displayName: string = "Group",
        isAllSilent: boolean = false,
        isInviteOnly: boolean = false,
        isPrivateChatAllowed: boolean = false
    ) {
        this.displayName = displayName;
        this.isAllSilent = isAllSilent;
        this.isInviteOnly = isInviteOnly
        this.isPrivateChatAllowed = isPrivateChatAllowed;
    }
}
